resteasy-core/src/main/java/org/jboss/resteasy/specimpl/BuiltResponse.java

```diff
@@ -262,8 +520,30 @@ public boolean hasEntity()
    public boolean bufferEntity()
    {
       abortIfClosed();
-      // no-op
-      return false;
+
+      if (bufferedEntity != null) return true;
+      if (streamRead) return false;
+      if (metadata.getFirst(HttpHeaderNames.CONTENT_TYPE) == null) return false;

```

Comment:
Please remove the media type check; there is no requirement that Content-Type header is available for bufferEntity to work